lib/ci.js

```diff
@@ -1,47 +1,50 @@
 'use strict'
 
+const util = require('util')
+const Arborist = require('@npmcli/arborist')
+const rimraf = util.promisify(require('rimraf'))
+
 const npm = require('./npm.js')
-const Installer = require('libcipm')
-const log = require('npmlog')
-const path = require('path')
-const pack = require('./pack.js')
-
-ci.usage = 'npm ci'
-
-ci.completion = (cb) => cb(null, [])
-
-module.exports = ci
-function ci (args, cb) {
-  const opts = {
-    // Add some non-npm-config opts by hand.
-    cache: path.join(npm.config.get('cache'), '_cacache'),
-    // NOTE: npm has some magic logic around color distinct from the config
-    // value, so we have to override it here
-    color: !!npm.color,
-    hashAlgorithm: 'sha1',
-    includeDeprecated: false,
-    log,
-    'npm-session': npm.session,
-    'project-scope': npm.projectScope,
-    refer: npm.referer,
-    dmode: npm.modes.exec,
-    fmode: npm.modes.file,
-    umask: npm.modes.umask,
-    npmVersion: npm.version,
-    tmp: npm.tmp,
-    dirPacker: pack.packGitDep
-  }
+const output = require('./utils/output.js')
+
+cmd.usage = 'npm ci'
+
+cmd.completion = (cb) => cb(null, [])
+
+module.exports = cmd
+function cmd(cb) {
+  ci()
+    .then(() => cb())
+    .catch(cb)
+}
 
-  for (const key in npm.config.list[0]) {
-    if (!['log', 'cache'].includes(key)) {
-      opts[key] = npm.config.list[0][key]
+async function ci () {
+  const where = npm.flatOptions.global
+    ? globalTop
```

Comment:
I think `npm ci` doesn't work for global packages, probably best to just throw an error if global is set, kind of like `audit` does.